$NetBSD: patch-ab,v 1.3 2010/11/22 05:54:17 dholland Exp $

- don't hardwire paths
- fix obvious bug handling XSLTPARAMS (PR 44126), already fixed upstream
- xmllint needs --nonet

--- xmlto.in.orig	2008-05-30 13:41:31.000000000 +0000
+++ xmlto.in
@@ -97,7 +97,7 @@ XSLTOPTS=
 SEARCHPATH=
 
 # Try to setup papersize using libpaper first ...
-if [ -x /usr/bin/paperconf ]
+if (type paperconf 1>/dev/null 2>&1)
 then
   papername=`paperconf -n`
   paperheight=`paperconf -mh | sed 's/ //g'`
@@ -128,7 +128,7 @@ EOF
   fi
 
 # ... or use magic paper size, based on LC_PAPER
-elif [ -x /usr/bin/locale ]
+elif (type locale 1>/dev/null 2>&1)
 then
   # For paper sizes we know about, specify them.
   h=$(locale LC_PAPER 2>/dev/null | head -n 1)
@@ -151,7 +151,7 @@ EOF
 fi
 
 # Magic encoding, based on locale
-if [ -x /usr/bin/locale ]
+if (type locale 1>/dev/null 2>&1)
 then
   charmap=$(locale charmap 2>/dev/null)
 
@@ -273,8 +273,8 @@ while [ "$#" -gt "0" ]; do
 	;;
   --stringparam)
 	MYPARAM="$2"
-	XSLTPARAMS="XSLTPARAMS --stringparam ${MYPARAM%=*}"
-	XSLTPARAMS="XSLTPARAMS $MYPARAM#*=}"
+	XSLTPARAMS="$XSLTPARAMS --stringparam ${MYPARAM%=*}"
+	XSLTPARAMS="$XSLTPARAMS $MYPARAM#*=}"
 	shift 2
 	;;
   --noclean)
@@ -427,8 +427,8 @@ if [ "$SKIP_VALIDATION" -eq 0 ] && [ "$S
 then
   VALIDATION="${XSLT_PROCESSED_DIR}/validation-errors"
   [ "$VERBOSE" -ge 1 ] && \
-    echo >&2 "xmllint >/dev/null --xinclude --postvalid \"$INPUT_FILE\""
-  xmllint >/dev/null --xinclude --postvalid "$INPUT_FILE" 2>"${VALIDATION}"
+    echo >&2 "xmllint >/dev/null --nonet --xinclude --postvalid \"$INPUT_FILE\""
+  xmllint >/dev/null --nonet --xinclude --postvalid "$INPUT_FILE" 2>"${VALIDATION}"
   xmllint_status=$?
   if [ $xmllint_status -ne 0 ]
   then
